doc/Hammer-Use/Hammer-Config.rst

```diff
@@ -0,0 +1,193 @@
+.. _config:
+
+The hammer-config library
+=======================================
+
+The `hammer-config library <https://github.com/ucb-bar/hammer/tree/master/src/hammer_config>`_ is the part of Hammer responsible for parsing Hammer YAML/JSON configuration files into Hammer IR. Hammer IR is used for the standardization and interchange of data between the different parts of Hammer and Hammer plugins.
+
+Basics
+------
+
+```
+foo:
+    bar:
+        adc: "yes"
+        dac: "no"
+```
+
+The basic idea of Hammer IR in YAML/JSON format is centered around a hierarchically nested tree of YAML/JSON dictionaries. For example, the above YAML snippet is translated to two variables which can be queried in code - `foo.bar.adc` would have `yes` and `foo.bar.dac` would have `no`.
+
+Overriding
+----------
+
+Hammer IR snippets frequently "override" each other. For example, a technology plugin might provide some defaults which a specific project can override with a project YAML snippet.
```

Comment:
This should be more specific (has been a huge source of confusion). We should note that there is a built-in order of precedence, which from lowest to highest (I think, though 2/3 might be backwards?) is 1) Hammer default, 2) tech plugin, 3) tool plugin, 4) User's Hammer IR. In 4), subsequent JSON/YAML files specified with `-p` in the command line have higher precedence, and keys appearing later if duplicated in a file also take precedence.

I think we discussed having a the debug json annotate the final source file for each key, but I can't find which issue it's in right now.